Server-errors-and-solutions.md

```diff
@@ -0,0 +1,80 @@
+This document talks about different types of server problems that have been fixed before, and it suggests ways to stop those problems from happening again. The types of problems include issues with data, problems when lots of people are using the server, situations that don't happen very often, making sure the server works with old internet browsers, mistakes in the way the server thinks, problems when the internet is slow, and errors that don't need to be written down. For each type of problem, it gives an example and tips on how to stop it from happening.
+
+# Categories
+
+* [Scalability issue](#scalability-issue)
+* [Corner cases](#corner-cases)
+* [Supporting old browsers](#supporting-old-browsers)
+* [Logic errors](#logic-errors)
+* [Data related issue](#data-related-issues)
+* [Race conditions / issues on slow networks](#race-conditions)
+* [Related to migration](#related-to-migrations)
+* [Errors that didn't need to be logged](#errors-that-didn't-need-to-be-logged)
+
+## Scalability Issue
+
+Issues that are related to features not scaling well on production. Example: (1) Exploration state data is too large(#12547) for some explorations and no further changes can be made to these explorations. (2) Suggestion models being fetched for review exceeds the query limit, so only an incomplete result is shown.
```

Comment:
Link properly to the issue in question. Make sure the link works.

Put a space before the parentheses.